Testnet/NFTStore/NFTStore/NFTStore.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Stratis.SmartContracts;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+[Deploy]
+public class NFTStore : SmartContract
+{
+    public SaleInfo GetSaleInfo(Address contract, ulong tokenId) => State.GetStruct<SaleInfo>($"SaleInfo:{contract}:{tokenId}");
+    private void SetSaleInfo(Address contract, ulong tokenId, SaleInfo value) => State.SetStruct($"SaleInfo:{contract}:{tokenId}", value);
+    private void ClearSaleInfo(Address contract, ulong tokenId) => State.Clear($"SaleInfo:{contract}:{tokenId}");
+
+    public NFTStore(ISmartContractState state)
+        : base(state)
+    {
+        EnsureNotPayable();
+    }
+
+    /// <summary>
+    /// Ensure the token is approved on the non fungible contract for store contract as pre-condition.
+    /// </summary>
+    public void Sale(Address contract, ulong tokenId, ulong price)
```

Comment:
Might save some gas on failures where `contract` is not actually a smart contract.

Assert(State.IsContract(contract)) would cost users a bit more gas every time but save failed transactions from attempting an invalid `TransferToken` call and 10k gas.

Reply:
I dont want that these check will cost the others who make things right.